features/puppet/defaults.pan

```diff
@@ -0,0 +1,53 @@
+template features/puppet/defaults;
+
+include 'quattor/functions/package';
+
+include 'components/puppet/config';
+
+'/software/components/puppet' = {
+
+    if(!is_defined(PUPPET_VERSION)){
+        # Default version is 3
+        version = '3.0.0';
+    }else{
+        if(match(PUPPET_VERSION, '^[1-9]$')){
+        # if just 1 figure, complete it to fill version number
+            version = PUPPET_VERSION + '.0.0';
+        }else{
+            version = PUPPET_VERSION;
+        };
+    };
+
+    # Defaults for version > 4
+    if(pkg_compare_version(version, '4.0.0') != PKG_VERSION_LESS){
+        SELF['hieraconf_file'] = '/etc/puppetlabs/code/environments/production/hiera.yaml';
```

Comment:
Try to avoid operating on `SELF` like this. It might actually be nicer to adapt this to instead include a defaults file for each major version, but it's up to you.

Perhaps something like:
```pan
include if (pkg_compare_version(PUPPET_VERSION, '4.0.0') != PKG_VERSION_LESS) 'features/puppet/defaults_4+';
include if (pkg_compare_version(PUPPET_VERSION, '5.0.0') != PKG_VERSION_LESS) 'features/puppet/defaults_5+';
```

```pan
template features/puppet/defaults_4+;

include 'components/puppet/config';

prefix '/software/components/puppet';
'hieraconf_file' = '/etc/puppetlabs/code/environments/production/hiera.yaml';
'puppet_cmd' = '/opt/puppetlabs/bin/puppet';
'puppetconf_file' = '/etc/puppetlabs/puppet/puppet.conf';
```


```pan
template features/puppet/defaults_5+;

include 'components/puppet/config';

prefix '/software/components/puppet';

'nodefiles/{quattor_default.pp}/contents' = "lookup('classes', Array[String], 'unique').include",

'modulepath' = '/etc/puppetlabs/code/environments/production/modules';

'nodefiles_path' = '/etc/puppetlabs/code/environments/production/manifests';

"hieraconf" = dict(
    "version", 5,
    "hierarchy", list(
        dict(
            "name", "quattor",
            "path", "quattor.yaml",
        )
    ),
    "defaults", dict(
        "datadir", "/etc/puppetlabs/code/environments/production/data",
        "data_hash", "yaml_data",
    ),
);

'hieradata_file' = '/etc/puppetlabs/code/environments/production/data/quattor.yaml';
```